### ISSUE TITLE
Errors on first run after clean install

### ISSUE THREAD

**Issue by lylemoffitt** [LightTable/LightTable#1790]
I was getting a bunch of errors every time I opened LT, so I decided to start fresh. After removing all related files and installing the newest version, I still get all of the same errors. This started the first time I opened LT, with no extra plugins or modifications.
### Steps to reproduce:
1.   Install LT
2.   Open LT
### Error description:

Console contains the following errors:

```
Error: Error: Command failed: /Users/Lyle/.zshrc: line 27: typeset: -U: invalid option
typeset: usage: typeset [-afFirtx] [-p] name[=value] ...
/usr/local/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh: line 37: typeset: -g: invalid option
typeset: usage: typeset [-afFirtx] [-p] name[=value] ...
/usr/local/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh: line 81: syntax error near unexpected token `always'
/usr/local/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh: line 81: `        } always {'
/Users/Lyle/.zshrc: line 81: `brew-man': not a valid identifier
is not ISeqable
    at Error (<anonymous>)
    at cljs.core.seq (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:2833:11)
    at cljs.core.empty_QMARK_ (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:3693:62)
    at eval (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:31483:32)
    at ChildProcess.exithandler (child_process.js:641:7)
    at ChildProcess.EventEmitter.emit (events.js:98:17)
    at maybeClose (child_process.js:735:16)
    at Socket.eval (child_process.js:950:11)
    at Socket.EventEmitter.emit (events.js:95:17)
    at Pipe.eval [as close] (net.js:466:12)
```

---

Included is the Instruments [trace](https://www.dropbox.com/sh/kr7l584p3wmoqdg/AAAAnJ-CdCEoNurj7SHexMpPa?dl=0) file, gathered during the opening of the application.

**Comment by lylemoffitt** [comment]
1.  I always open the application, never from the CLI. Trying the latter produces the same results.
2.  Commenting out line 27 of `.zshrc` removes only the associated error. The rest is the same beginning with.. 

    ```
Error: Error: Command failed: /usr/local/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh: line 37: typeset: -g: invalid option
typeset: usage: typeset [-afFirtx] [-p] name[=value] ...
    ```
3.   The above error (specifically regarding line 37) can be silenced by removing the related `source [...]` command.
4.   The last error (referring to line 81) is silenced by changing the line as follows:

    ```
brew-man() { man brew | less -p "^       $1 "; }     ## before ##
brewman() { man brew | less -p "^       $1 "; }     ## after ##
    ```
5.  The reason why LT is picking up `.zshrc` in the first place is that my `.profile` file `source`s it.



----

In conclusion, after silencing all error I went back to see if uncommenting the previous two (those regarding 27 and 37) would reproduce them. To my surprise, they did not. The only change that was necessary (it seems) was to change the name of the function `brew-man` so as to remove the hyphen. 

I would be curious to know if either of you can reproduce this error with a similarly named shell function in your `.zshrc` file. It seems rather unlikely to me that such a thing could be specific to my system. I would wager a string parsing bug somewhere is more likely.

Lastly, as a further potential missing piece of the puzzle, it might help to know that my login shell is also zsh (rather than the default bash). It's not a common choice among OSX users, and it may help explain why ~.zs

**Comment by Mouvedia** [comment]
For what I can gather*, `brew-man` wouldn't be a valid name.

http://unix.stackexchange.com/a/168222/7579
